### ISSUE TITLE
Delegate conversion for extension methods contain wrong type error messages when omitting type arguments

### ISSUE THREAD

**Issue by OmarTawfik** [dotnet/roslyn#16041]
**Repro:**

```csharp
public static class FooExtensions
{
    public static T ExtensionMethod1<T>(this object obj) => default(T);
}
public class Class1
{
    public void Test()
    {
        System.Func<object> delegateConversion1 = "literal".ExtensionMethod1<>;
    }
}
```

**Error Displayed:**
`Error CS0407: '? FooExtensions.ExtensionMethod1<?>(object)' has the wrong return type`

**Expected:**
Display a more meaningful error instead of using `?`

Discovered by #15966

**Comment by CyrusNajmabadi** [comment]
This will not give you `CS8389: Omitting the type argument is not allowed in the current context` in an upcoming release of C#.  Fixed with https://github.com/dotnet/roslyn/pull/81495
